src/modules/status-list/status-list.controller.ts

```diff
@@ -0,0 +1,156 @@
+    const issuer = DID.from(credential.issuer).did;
+
+    if (issuer !== currentUser) {
+      throw new ForbiddenException(
+        'Requester is not the issuer of given credential'
+      );
+    }
+
+    await this.roleService.verifyEnrolmentIssuer({
+      issuerDID: issuer,
+      claimType: namespace,
+    });
+
+    return await this.statusListService.addStatusListEntry(credential);
+  }
+
+  /*
+   * Returns the StatusList2021Credential with the bit associated with the provided VerifiableCredential marked
+   * as true.
+   * The updated StatusList2021Credential is not signed and not persisted.
+   * https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021credential
+   *
+   * The operation is conducted server-side (rather than client-side) to encourage
+   * clients to register their updates with the server, as this maybe required in future versions.
+   * (For example, for the implementation of a many-to-one credential to status-list relationship).
+   */
+  @Auth()
+  @Post('/credentials/status/initiate')
+  @ApiCreatedResponse({ type: StatusListCredentialDto })
+  @ApiForbiddenResponse({ description: 'User is not authorized to revoke.' })
+  @ApiConflictResponse({ description: 'Credential already revoked.' })
+  async initiateStatusUpdate(
+    @User() currentUser: string,
+    @Body() { verifiableCredential }: RegisterRevokeInputDto
+  ): Promise<StatusListCredentialDto> {
+    await Promise.all([
+      this.roleService.verifyEnrolmentIssuer({
+        issuerDID: DID.from(verifiableCredential.issuer).did,
+        claimType: verifiableCredential.credentialSubject.role.namespace,
+      }),
+      this.statusListService.verifyCredential(verifiableCredential),
+      this.roleService.verifyRevoker({
+        revokerDID: currentUser,
+        claimType: verifiableCredential.credentialSubject.role.namespace,
+      }),
+    ]);
+
+    return await this.statusListService.markStatusListCredential(
+      verifiableCredential,
+      currentUser
+    );
+  }
+
+  @Auth()
+  @Post('/credentials/status/finalize')
+  @ApiCreatedResponse({ type: StatusListVerifiableCredentialDto })
+  @ApiForbiddenResponse({ description: 'User is not authorized to revoke.' })
+  @ApiBadRequestResponse()
+  async persistUpdate(
+    @Body() { statusListCredential }: SignRevokeInputDto
+  ): Promise<StatusListVerifiableCredentialDto> {
+    const issuer = DID.from(statusListCredential.issuer).did;
+    const revokedCredentialId = statusListCredential.credentialSubject.id;
+
+    const statusList = await this.statusListService.getCredential(
+      revokedCredentialId
+    );
+
+    if (!statusList) {
+      throw new BadRequestException('Credential was not registered');
+    }
+
+    await Promise.all([
+      this.statusListService.verifyCredential(statusListCredential),
+      this.roleService.verifyRevoker({
+        revokerDID: issuer,
+        claimType: statusList.namespace,
+      }),
+    ]);
```

Comment:
I think this won't scale if we go to many-to-one credentials as the `revokedCredentialId = statusListCredential.credentialSubject.id` relationship won't hold. I think we should ask the client to provide the namespace and then search through the known status list ids for that namespace. See pseudocode here: https://github.com/energywebfoundation/ssi-hub/blob/status-list-2022-implementation/docs/revocation.md#update-entry-in-a-status-list

I think this is better than asking the client to provide the `verifiableCredential` again as they may want to perform a bulk update.